### ISSUE TITLE
extraOptions is ignored when using awsx.ecr.Image

### ISSUE THREAD

**Issue by rlukata** [pulumi/pulumi-awsx#1148]
### What happened?

extraOptions is ignored when using awsx.ecr.Image

### Example

```
const image = new awsx.ecr.Image(`docker-image`, {
        repositoryUrl: repo.repository.repositoryUrl,
        context: context,
        dockerfile: dockerFile,
        platform: 'linux/amd64',
        builderVersion: awsx.ecr.BuilderVersion.BuilderBuildKit,
        extraOptions: ['--quiet']
    });
```

### Output of `pulumi about`

```
CLI
Version      3.86.0
Go Version   go1.21.1
Go Compiler  gc

Plugins
NAME            VERSION
aws             5.42.0
aws             5.42.0
aws-apigateway  1.0.1
awsx            2.1.1
command         0.8.2
docker          4.4.4
nodejs          unknown
tls             4.10.0

Host
OS       darwin
Version  14.1
Arch     arm64
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by mikhailshilkov** [comment]
It turns out that `extraOptions` are supported in pulumi-docker V4 anymore: https://github.com/pulumi/pulumi-docker/issues/424

We should have deprecated/removed the `extraOptions` on ecr:Image. We should still probably do it, since we aren't breaking any existing V2 behavior (V1 behavior is already broken in the new major version). @danielrbradley @thomas11 what do you think?

**Comment by thomas11** [comment]
> It turns out that `extraOptions` are not supported in pulumi-docker V4 anymore: [pulumi/pulumi-docker#424](https://github.com/pulumi/pulumi-docker/issues/424)
> 
> We should have deprecated/removed the `extraOptions` on ecr:Image. We should still probably do it, since we aren't breaking any existing V2 behavior (V1 behavior is already broken in the new major version). @danielrbradley @thomas11 what do you think?

@mikhailshilkov yes, I think removing something that isn't working can be considered a non-breaking change, we have done this before.
